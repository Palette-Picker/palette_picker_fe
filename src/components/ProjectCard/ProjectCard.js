import React from 'react';
import './ProjectCard.scss';
import close from '../../images/close-icon-svg-4.jpg'

const ProjectCard = ({ project, handleModal, handleKeyPress, handleProjectDelete }) => {
  const { palettes } = project;
  const paletteCards = palettes.map(palette => {
    const { color1, color2, color3, color4, color5 } = palette;
    const paletteUnlocked = [
      {
        color1,
        isLocked: false
      },
      {
        color2,
        isLocked: false
      },
      {
        color3,
        isLocked: false
      },
      {
        color4,
        isLocked: false
      },
      {
        color5, 
        isLocked: false
      }  
    ]
    return <section 
      key={palette.id}
      className='palette'
      >
      <h4>{palette.name}</h4>
      <div
        className='five-colors'
        onClick={(e) => handleModal(e, paletteUnlocked, { id: project.id, name: project.name }, palette.name, palette.id)}
      >
        <div 
          className='palette-color'
          style={{backgroundColor: `${color1}`}}
        ></div>
        <div 
          className='palette-color'
          style={{backgroundColor: `${color2}`}}
        ></div>
        <div 
          className='palette-color'
          style={{backgroundColor: `${color3}`}}
        ></div>
        <div 
          className='palette-color'
          style={{backgroundColor: `${color4}`}}
        ></div>
        <div 
          className='palette-color'
          style={{backgroundColor: `${color5}`}}
        ></div>
      </div>
    </section>
  })
  return (
    <article className='project-card'>
      <img className='img--close' src={close} onClick={() => handleProjectDelete(project.id)} />
      <h2 contenteditable="true" onKeyPress={(e) => handleKeyPress(e, project.id, e.target.innerText)}>{project.name}</h2>
      {paletteCards}
    </article>
  )

};

export default ProjectCard;