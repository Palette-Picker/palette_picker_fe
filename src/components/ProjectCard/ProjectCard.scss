.palette-color {
  height: 50px;
  width: 50px;
}