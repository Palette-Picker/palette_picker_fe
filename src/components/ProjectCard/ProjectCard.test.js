import React from 'react';
import { shallow } from 'enzyme';
import ProjectCard from './ProjectCard';

describe('ProjectCard', () => {

  let wrapper;
  const mockProject = 
    {
      id: 19,
      name: "Colors",
      palettes: [
        {
          id: 29,
          name: "Purples and greens",
          color1: "#82d173",
          color2: "#abfaa9",
          color3: "#95a3b3",
          color4: "#4c2c69",
          color5: "#42253b",
          project_id: 19,
          created_at: "2019-12-09T00:11:51.362Z",
          updated_at: "2019-12-09T00:11:51.362Z"
        }]
    };

  const mockPass = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <ProjectCard
        project={mockProject}
        passPaletteNameAndColors={mockPass}
      />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call passPaletteNameAndColors when clicked', () => {
    wrapper.find('div').at(0).simulate('click');
    // expect(mockPass).toHaveBeenCalledWith(paletteUnlocked, { 19, 'Colors'}, 'Purples and greens')
    expect(mockPass).toHaveBeenCalled();
  });

});
