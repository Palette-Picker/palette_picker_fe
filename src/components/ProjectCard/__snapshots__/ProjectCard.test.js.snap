// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard should match the snapshot 1`] = `
<article
  className="project-card"
>
  <h2>
    Colors
  </h2>
  <section
    className="palette"
    key="29"
  >
    <h4>
      Purples and greens
    </h4>
    <div
      className="five-colors"
      onClick={[Function]}
    >
      <div
        className="palette-color"
        style={
          Object {
            "backgroundColor": "#82d173",
          }
        }
      />
      <div
        className="palette-color"
        style={
          Object {
            "backgroundColor": "#abfaa9",
          }
        }
      />
      <div
        className="palette-color"
        style={
          Object {
            "backgroundColor": "#95a3b3",
          }
        }
      />
      <div
        className="palette-color"
        style={
          Object {
            "backgroundColor": "#4c2c69",
          }
        }
      />
      <div
        className="palette-color"
        style={
          Object {
            "backgroundColor": "#42253b",
          }
        }
      />
    </div>
  </section>
</article>
`;
